src/map.cpp

```diff
@@ -1703,6 +1703,96 @@ bool map::moppable_items_at(const int x, const int y)
     return false;
 }
 
+void map::decay_fields_and_scent( const int amount )
+{
+    // Decay scent separately, so that later we can use field count to skip empty submaps
+    for( int x = 0; x < my_MAPSIZE * SEEX; x++ ) {
```

Comment:
This has better cache locality anyway, bouncing back and forth between the scent map and the fields will slow things down.